google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/AbstractConnectionImplTest.java

```diff
@@ -735,6 +735,22 @@ public void testGetCommitTimestamp() {
     }
   }
 
+  @Test
+  public void testGetCommitResponse() {
+    try (Connection connection = getConnection()) {
+      if (isGetCommitTimestampAllowed()) {
```

Comment:
if blocks in tests are code smells. Only one path is taken and tested. Does isGetCommitTimestampAllowed() return true or false here?

Reply:
This and the other methods in this abstract test class are used to test that a `Connection` responds correctly in all possible states that it can be. The specific state of a `Connection` is implemented in a concrete subclass, and that determines whether `isGetCommitTimestampAllowed()` will return `true` or `false`.

Example 1: [ConnectionImplAutocommitReadWriteNoActionsTest](ConnectionImplAutocommitReadWriteNoActionsTest) verifies that a connection that is in autocommit read/write mode and has not yet executed any statements, responds correctly to the state-dependent methods. [isGetCommitTimestampAllowed()](https://github.com/googleapis/java-spanner/blob/f9ac29cd05e775c8ea17e0e227eb925df88a8dba/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplAutocommitReadWriteTest.java#L140) returns false, as the connection has not executed a commit. `getCommitResponse()` should therefore not return a result / throw an exception.

Example 2: [ConnectionImplAutocommitReadWriteAfterTemporaryTransactionTest](https://github.com/googleapis/java-spanner/blob/f9ac29cd05e775c8ea17e0e227eb925df88a8dba/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplAutocommitReadWriteTest.java#L894) verifies that a connection in autocommit read/write mode that has just successfully executed an ad-hoc transaction does return a commit response. [isGetCommitTimestampAllowed()](https://github.com/googleapis/java-spanner/blob/f9ac29cd05e775c8ea17e0e227eb925df88a8dba/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplAutocommitReadWriteTest.java#L1004) therefore returns true for this implementation.

This set of test classes is used to generate a [SQL script](https://github.com/googleapis/java-spanner/blob/master/google-cloud-spanner/src/test/resources/com/google/cloud/spanner/connection/ConnectionImplGeneratedSqlScriptTest.sql) that can also be used for the same Connection API in other client languages.